File: src/test/programmers/java/level1/AddEumYang.java
package test.programmers.java.level1;

public class AddEumYang {
    public static void main(String[] args) {
        int[] arr = {4,7,12};
        boolean[] b = {true, false, true};
        System.out.println(solution(arr, b));
    }

    static int solution(int[] absolutes, boolean[] signs) {
        int answer = 0;

        for(int i = 0; i < absolutes.length; i++) {
            absolutes[i] *= signs[i] ? 1 : -1;
            answer += absolutes[i];
        }

        return answer;
    }
}

File: src/test/programmers/python/level1/AddEumYang.py
def solution(a, b):
    answer = 0
    for i, j in zip(a, b):
        answer += (i * j)
    return answer

# def solution(absolutes, signs):
#     return sum(absolute if sign else -absolute for absolute, sign in zip(absolutes, signs))

# print(solution([4,7,12], [True,False,True]))
